managing-large-amounts-of-file-permissions.md

```diff
@@ -0,0 +1,17 @@
+# Managing large amounts of file permissions
+
+## Summary
+
+Users should not use the `recurse => true` attribute on the file resource because it is non-performant on the agent and downstream when it places 100s or 1000s of file resources in a report which then needs to be ingested by PuppetDB.  
```

Comment:
Should we limit this to only certain cases, like this:

> Users should not use the `recurse => true` attribute on the file resource **when the goal is to recursively manage permissions** ...

Reply:
are you saying there is a valid reason to use recurse => true?

Reply:
Yeah, when you want to copy an entire folder of stuff from a `puppet://` or HTTP source. Right?

Reply:
Yea, I think we're in agreement that the problem statement and the solution aren't in alignment.  

However, I think instead of constraining the problem we should list the various ways you can avoid using `recurse => true`.  This is essentially the "don't use `recurse => true`" best practice but I know @reidmv would tell me I can't name my best practice that way.

Reply:
How about renaming the file to `files-and-recurse-true.md`, or `alternatives-to-recursive-file-resources.md`, or even just `recursive-file-resources.md` ?

Then we can generalize this best practice by adding more bad-use cases and recommended alternatives.

Reply:
yea I like that